jenkins-integration/jenkins-jobs/scenarios/couch-environment-timeout/Jenkinsfile

```diff
@@ -0,0 +1,97 @@
+     code_deploy: [
+             type: "r10k",
+             control_repo: "git@github.com:puppetlabs/puppetlabs-puppetserver_perf_control.git",
+             basedir: "/etc/puppetlabs/code-staging/environments",
+             environments: ["production"],
+             hiera_config_source_file: "/etc/puppetlabs/code-staging/environments/production/root_files/hiera.yaml"
+     ],
+     server_java_args: "-Xms12g -Xmx12g",
+     puppet_settings: [
+             main: [
+                     "environment_timeout": "unlimited"
+             ],
+             master: [
+                     "environment_timeout": "unlimited"
+             ]
+     ],
+     background_scripts: [
+             "./jenkins-jobs/common/scripts/background/curl-server-metrics-loop.sh"
+     ],
+     archive_sut_files: [
+             "/var/log/puppetlabs/puppetserver/metrics.json"
+     ]
+    ]
+}
+
+environment_timeout_0_500 = environment_timeout_0_base_config()
+environment_timeout_0_500.put('job_name', 'pe-couch-environment-timeout-0-500')
+environment_timeout_0_500.put('gatling_simulation_config', '../simulation-runner/config/scenarios/pe-couch-medium-500-2-hours.json')
+
+environment_timeout_unlimited_500 = environment_timeout_unlimited_base_config()
+environment_timeout_unlimited_500.put('job_name', 'pe-couch-environment-timeout-unlimited-500')
+environment_timeout_unlimited_500.put('gatling_simulation_config', '../simulation-runner/config/scenarios/pe-couch-medium-500-2-hours.json')
+
+environment_timeout_0_1000 = environment_timeout_0_base_config()
+environment_timeout_0_1000.put('job_name', 'pe-couch-environment-timeout-0-1000')
+environment_timeout_0_1000.put('gatling_simulation_config', '../simulation-runner/config/scenarios/pe-couch-medium-1000-2-hours.json')
+
+environment_timeout_unlimited_1000 = environment_timeout_unlimited_base_config()
+environment_timeout_unlimited_1000.put('job_name', 'pe-couch-environment-timeout-unlimited-1000')
+environment_timeout_unlimited_1000.put('gatling_simulation_config', '../simulation-runner/config/scenarios/pe-couch-medium-1000-2-hours.json')
+
+environment_timeout_0_1250 = environment_timeout_0_base_config()
+environment_timeout_0_1250.put('job_name', 'pe-couch-environment-timeout-0-1250')
+environment_timeout_0_1250.put('gatling_simulation_config', '../simulation-runner/config/scenarios/pe-couch-medium-1250-2-hours.json')
+
+environment_timeout_unlimited_1250 = environment_timeout_unlimited_base_config()
+environment_timeout_unlimited_1250.put('job_name', 'pe-couch-environment-timeout-unlimited-1250')
+environment_timeout_unlimited_1250.put('gatling_simulation_config', '../simulation-runner/config/scenarios/pe-couch-medium-1250-2-hours.json')
+
+pipeline.multipass_pipeline([
+        environment_timeout_0_500,
+        environment_timeout_unlimited_500,
+        environment_timeout_0_1000,
+        environment_timeout_unlimited_1000,
+        environment_timeout_0_1250,
+        environment_timeout_unlimited_1250,
+])
```

Comment:
I'd be open to doing something like that.

I copied this approach pretty much verbatim from the [static-catalogs Jenkinsfile](https://github.com/puppetlabs/gatling-puppet-load-test/blob/3f57c276b9d1cc9bdc0c1c0297974be914921a36/jenkins-integration/jenkins-jobs/scenarios/couch-static-catalogs/Jenkinsfile).  I think @cprice404 went with this approach as a compromise from having to repeat the entire config for each variation but also was leery about having Groovy code that was too sophisticated, like if Jenkins might have security constraints or interpreter problems or something?  That said, this could cut down on the repetition a bit and, if we like it better, could be something we model for all other jobs like this.

Guess I'm on the fence a bit about this, leaning a bit toward not doing it until we have at least one more variation of this kind of job.
@cprice404 - what do you think?